src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -262,6 +262,8 @@ private BoundExpression BindEventAssignment(AssignmentExpressionSyntax node, Bou
             }
             else
             {
+                CheckImplicitThisCopyInReadOnlyMember(receiverOpt, method, diagnostics);
```

Comment:
>CheckImplicitThisCopyInReadOnlyMember(receiverOpt, method, diagnostics); [](start = 16, length = 72)

It doesn't look like this is going to warn on an event of another struct stored in a field of the current type. Please make sure there is a test for this scenario

Reply:
This is by design (unfortunately). See #33968. I can add a test to demonstrate the existing behavior though.

Reply:
> This is by design (unfortunately). See #33968. I can add a test to demonstrate the existing behavior though.

It looks like the issue is talking about a different scenario involving readonly fields in a non-readonly member

---
In reply to: [284909657](https://github.com/dotnet/roslyn/pull/35710#discussion_r284909657) [](ancestors = 284909657)

Reply:
Let's sync up about this offline tomorrow.